docs/OBJECT_LOCK_TEST_PLAN.md

```diff
@@ -0,0 +1,157 @@
+# Object Lock Feature Test Plan
+
+## Feature Component Description
+
+Implementing Object Lock will introduce six new APIs:
+
+- putObjectLockConfiguration
+- getObjectLockConfiguration
+- putObjectRetention
+- getObjectRetention
+- putObjectLegalHold
+- getObjectLegalHold
+
+Along with these APIs, putBucket, putObject, deleteObject, and multiObjectDelete
+be affected. In Arsenal, both the BucketInfo and ObjectMD models will be
+updated. Bucket policy and IAM policy permissions will be updated to include
+the new API actions.
+
+## Functional Tests
```

Comment:
It'd be good to add functional tests to `getObject` and `copyObject` as well, e.g:
- `getObject` returns the object with `ObjectLockMode` and `ObjectLockRetainUntilDate` if the object has valid object lock configuration in place.
- `copyObject` should copy the object with its object lock configuration if the object has any.